internal/jobservice/repository/sql_job_service.go

```diff
@@ -0,0 +1,266 @@
+package repository
+
+import (
+	"database/sql"
+	"fmt"
+	"sync"
+	"time"
+
+	_ "modernc.org/sqlite"
+
+	"github.com/G-Research/armada/internal/jobservice/configuration"
+	js "github.com/G-Research/armada/pkg/api/jobservice"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Internal structure for storing in memory JobTables and Subscription JobSets
+// Locks are used for concurrent access of map
+type JobStatus struct {
+	jobMap        map[string]*JobTable
+	jobLock       sync.RWMutex
+	subscribeMap  map[string]*SubscribeTable
+	subscribeLock sync.RWMutex
+}
+
+func NewJobStatus(jobMap map[string]*JobTable, subscribeMap map[string]*SubscribeTable) *JobStatus {
+	return &JobStatus{jobMap: jobMap, subscribeMap: subscribeMap}
+}
+
+// SQLJobService for persisting to DB.
+type SQLJobService struct {
+	jobStatus        *JobStatus
+	jobServiceConfig *configuration.JobServiceConfiguration
+	db               *sql.DB
+}
+
+func NewSQLJobService(jobMap *JobStatus, config *configuration.JobServiceConfiguration, db *sql.DB) *SQLJobService {
+	return &SQLJobService{jobStatus: jobMap, jobServiceConfig: config, db: db}
+}
+
+type SubscribedTuple struct {
+	Queue  string
+	JobSet string
+}
+
+// Create a Table from a hard-coded schema.
+// Open to suggestions on how to make this better
+func (s *SQLJobService) CreateTable() {
+	_, err := s.db.Exec("DROP TABLE IF EXISTS jobservice")
+	if err != nil {
+		panic(err)
+	}
+	_, err = s.db.Exec(`
+CREATE TABLE jobservice (
+Queue TEXT,
+JobSetId TEXT,
```

Comment:
I just thought of this - of course this may be completely "premature optimization", but any chance you might need indexes on JobSetId and/or JobId, if you're doing queries/updates on those, and the cardinality (num rows) of this table gets big?  (If it's less than a few thousand rows, then indexes aren't worth the bother.)

Reply:
I'm pretty sure we'll want indexes

Reply:
I'll add some indexes.

Reply:
Please take a look!  I marked that JOB_ID must be a unique id and added an index for queue and job-set.

Reply:
I think the index is good- I don't think you need the constraint, rather you should define `JobId` to be the primary key as per https://www.sqlitetutorial.net/sqlite-primary-key/.  That will essentially define it is unique and indexed.